spring-data-mongodb/src/test/java/org/springframework/data/mongodb/repository/query/StringBasedMongoQueryUnitTests.java

```diff
@@ -293,10 +303,36 @@ public void shouldSupportExpressionsInCustomQueries() {
 		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
 	}
 
+	@Test // GH-3050
+	public void shouldSupportExpressionsAndPropertiesInCustomQueries() {
+
+		ConvertingParameterAccessor accessor = StubParameterAccessor.getAccessor(converter, "Matthews");
+		StringBasedMongoQuery mongoQuery = createQueryForMethod("findByQueryWithExpressionAndProperty", String.class);
+
+		org.springframework.data.mongodb.core.query.Query query = mongoQuery.createQuery(accessor);
+		org.springframework.data.mongodb.core.query.Query reference = new BasicQuery(
+				"{'lastname' : 'Matthews', 'firstname' : 'some-default'}");
+
+		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
+	}
+
+	@Test // GH-3050
+	public void shouldSupportPropertiesInCustomQueries() {
+
+		properties.put("foo", "bar");
+		ConvertingParameterAccessor accessor = StubParameterAccessor.getAccessor(converter);
+		StringBasedMongoQuery mongoQuery = createQueryForMethod("findByQueryWithProperty");
+
+		org.springframework.data.mongodb.core.query.Query query = mongoQuery.createQuery(accessor);
+		org.springframework.data.mongodb.core.query.Query reference = new BasicQuery("{'lastname' : 'bar'}");
+
+		assertThat(query.getQueryObject()).isEqualTo(reference.getQueryObject());
+	}
+
 	@Test // DATAMONGO-1244
 	public void shouldSupportExpressionsInCustomQueriesWithNestedObject() {
 
-		ConvertingParameterAccessor accessor = StubParameterAccessor.getAccessor(converter, true, "param1", "param2");
```

Comment:
why's `param2` gone?

Reply:
because the method declares 2 parameters and previously we passed in 3 arguments.